javascript/ql/src/semmle/javascript/dataflow/TaintTracking.qll

```diff
@@ -809,6 +810,66 @@ module TaintTracking {
 
   }
 
+  /**
+   * A function that returns the result of a sanitizer check.
+   */
+  private class SanitizingFunction extends Function {
+    Parameter sanitizedParameter;
+
+    TaintTracking::SanitizerGuardNode sanitizer;
+
+    boolean sanitizerOutcome;
+
+    SanitizingFunction() {
+      exists(Expr e |
+        exists(SsaExplicitDefinition ssa |
+          ssa.getDef().getSource() = sanitizer.asExpr() and
+          ssa.getVariable().getAUse() = getAReturnedExpr()
+        )
+        or
+        sanitizer.asExpr() = getAReturnedExpr()
+        |
+        DataFlow::parameterNode(sanitizedParameter).flowsToExpr(e) and
+        sanitizer.sanitizes(sanitizerOutcome, e)
+      ) and
+      getNumParameter() = 1 and
+      sanitizedParameter = getParameter(0)
+    }
+
+    /**
+     * Holds if this function sanitizes argument `e` of call `call`, provided the call evaluates to `outcome`.
+     */
+    predicate isSanitizingCall(DataFlow::CallNode call, Expr e, boolean outcome) {
+      exists(DataFlow::Node arg |
+        arg.asExpr() = e and
+        arg = call.getArgument(0) and
+        call.getNumArgument() = 1 and
+        FlowSteps::argumentPassing(call, arg, this, sanitizedParameter) and
+        outcome = sanitizerOutcome
+      )
+    }
+
+    /**
+     * Holds if this function applies to the flow in `cfg`.
+     */
+    predicate appliesTo(Configuration cfg) {
+      not sanitizer instanceof AdditionalSanitizerGuardNode or
```

Comment:
Shouldn't this be `cfg.isSanitizerGuard(sanitizer)`?

Reply:
Fixed. `cfg.isBarrierGuard(sanitizer)` is the right thing to use.